pkg/cmd/util/clientcmd/factory.go

```diff
@@ -45,12 +47,58 @@ func New(flags *pflag.FlagSet) *Factory {
 
 	// TODO: there should be two client configs, one for OpenShift, and one for Kubernetes
 	clientConfig := DefaultClientConfig(flags)
+	clientConfig = defaultingClientConfig{clientConfig}
 	f := NewFactory(clientConfig)
 	f.BindFlags(flags)
 
 	return f
 }
 
+// defaultingClientConfig detects whether the provided config is the default, and if
+// so returns an error that indicates the user should set up their config.
+type defaultingClientConfig struct {
+	nested kclientcmd.ClientConfig
+}
+
+// RawConfig calls the nested method
+func (c defaultingClientConfig) RawConfig() (kclientcmdapi.Config, error) {
+	return c.nested.RawConfig()
+}
+
+// Namespace calls the nested method
+func (c defaultingClientConfig) Namespace() (string, bool, error) {
+	return c.nested.Namespace()
+}
+
+// ClientConfig returns a complete client config
+func (c defaultingClientConfig) ClientConfig() (*kclient.Config, error) {
+	cfg, err := c.nested.ClientConfig()
+	if err != nil {
+		if kclientcmd.IsEmptyCluster(err) {
```

Comment:
More invasive change, also it would result in a <carry> because Kube would
have different messages than us.

On Thu, Nov 5, 2015 at 11:07 AM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/util/clientcmd/factory.go
> <https://github.com/openshift/origin/pull/5722#discussion_r44029872>:
>
> > +
> > +// RawConfig calls the nested method
> > +func (c defaultingClientConfig) RawConfig() (kclientcmdapi.Config, error) {
> > +	return c.nested.RawConfig()
> > +}
> > +
> > +// Namespace calls the nested method
> > +func (c defaultingClientConfig) Namespace() (string, bool, error) {
> > +	return c.nested.Namespace()
> > +}
> > +
> > +// ClientConfig returns a complete client config
> > +func (c defaultingClientConfig) ClientConfig() (*kclient.Config, error) {
> > +	cfg, err := c.nested.ClientConfig()
> > +	if err != nil {
> > +		if kclientcmd.IsEmptyCluster(err) {
>
> Why would I be inspecting and transforming the error here instead of
> CheckErr?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5722/files#r44029872>.
>